pkg/adapters/grpc/event_grpc.go

```diff
@@ -28,7 +30,19 @@ type EventServiceServer struct {
 	eventPlugin events.EventService
 }
 
+func (s *EventServiceServer) checkPluginRegistered() error {
+	if s.eventPlugin == nil {
+		return status.Errorf(codes.Unimplemented, "Event plugin not registered")
```

Comment:
Can you use the new error function to here.

Reply:
done.